course/serializers.py

```diff
@@ -100,11 +101,11 @@ def create(self, validated_data):
                     start_datetime=start_datetime,
                     end_datetime=end_datetime,
                     instructor=course.instructor,
-                    is_confirmed=True
+                    is_confirmed= course.is_confirmed and end_datetime <= confirmed_end_date
```

Comment:
you're currently comparing a datetime against a date, shouldn't this be current_date <= confirmed_end_date?

Reply:
ack